File: leet-code/go/0001 two-sum.go
package leet

func twoSum(nums []int, target int) []int {
	numsM := map[int]int{}
	out := make([]int, 2)

	for i, num := range nums {
		if ix, ok := numsM[target-num]; ok {
			return []int{ix, i}
		}

		numsM[num] = i
	}

	return out
}

File: leet-code/python/0001_two_sum.py
class Solution:
  def twoSum(self, nums: List[int], target: int) -> List[int]:
    m = {}
    for i in range(len(nums)):
      if nums[i] in m:
        return [m[nums[i]], i]
      
      m[target - nums[i]] = i
    
    return []